pkg/iac/scanners/terraform/parser/parser_test.go

```diff
@@ -2854,3 +2854,26 @@ locals {
 	attr := bucket.GetAttribute("test")
 	require.Equal(t, "some_value", attr.Value().AsString())
 }
+
+func Test_UnknownModuleSource(t *testing.T) {
+	files := map[string]string{
+		`main.tf`: `
+module "test" {
+  source = "${foo}/modules/foo"
```

Comment:
Is this valid to do? Regardless we shouldn't panic so this PR makes sense but just curious.